mailbox/api/src/main/java/org/apache/james/mailbox/extractor/ContentTypeFilteringTextExtractor.java

```diff
@@ -0,0 +1,49 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.extractor;
+
+import java.io.InputStream;
+import java.util.Objects;
+
+public class ContentTypeFilteringTextExtractor implements TextExtractor {
```

Comment:
This class should not be part of mailbox API as it is a (generic) implementation.

It would fit better in a lib maven module.

But, similary to what we did with the CachingTextExtractor, we can temporary locate it in tika module...